Asgard/Communications/Classes/CbusEventManager.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Asgard.Data;
+using Asgard.Data.Interfaces;
+using Microsoft.Extensions.Logging;
+
+namespace Asgard.Communications
+{
+    public class CbusEventManager :
+        ICbusEventManager,
+        IDisposable
+    {
+        #region Fields
+
+        private readonly ICbusMessenger cbusMessenger;
+
+        private readonly ILogger logger;
+
+        private readonly ConcurrentDictionary<CbusEventKey, bool?> cbusEventStates = new();
+        private readonly ConcurrentDictionary<CbusEventKey, CbusEventCallback> cbusEventCallbacks = new();
+        private readonly EmptyingConcurrentQueue<Exception> cbusEventExceptionQueue = new();
+
+        private readonly CancellationTokenSource cancellationTokenSource = new();
+
+        #endregion
+
+        #region Properties
+
+        protected CancellationToken CancellationToken => this.cancellationTokenSource.Token;
+
+        #endregion
+
+        #region Constructors
+
+        public CbusEventManager(ICbusMessenger cbusMessenger, ILogger logger)
+        {
+            this.cbusMessenger = cbusMessenger;
+            this.logger = logger;
+
+            Task.Run(() => ProcessQueue());
```

Comment:
Would be better to kick this off in another method rather than in the constructor

Reply:
All it should do is hit the manual-reset-event and simply wait for it to be reset, or cancelled. The manager has no way of knowing when it will be receiving incoming events, so I'm thinking it should have its queue handling started as soon as possible.